File: briar-headless/src/main/java/org/briarproject/briar/headless/BriarHeadlessApp.java
package org.briarproject.briar.headless;

import org.briarproject.bramble.BrambleCoreEagerSingletons;
import org.briarproject.bramble.BrambleCoreModule;
import org.briarproject.bramble.account.HeadlessAccountModule;
import org.briarproject.bramble.system.DesktopSecureRandomModule;
import org.briarproject.briar.BriarCoreEagerSingletons;
import org.briarproject.briar.BriarCoreModule;

import javax.inject.Singleton;

import dagger.Component;

@Component(modules = {
		BrambleCoreModule.class,
		BriarCoreModule.class,
		DesktopSecureRandomModule.class,
		HeadlessAccountModule.class,
		HeadlessModule.class
})
@Singleton
public interface BriarHeadlessApp
		extends BrambleCoreEagerSingletons, BriarCoreEagerSingletons {
	Router router();
}

File: briar-headless/src/main/java/org/briarproject/briar/headless/BriarHeadlessApp.kt
package org.briarproject.briar.headless

import dagger.Component
import org.briarproject.bramble.BrambleCoreEagerSingletons
import org.briarproject.bramble.BrambleCoreModule
import org.briarproject.bramble.BrambleJavaEagerSingletons
import org.briarproject.bramble.BrambleJavaModule
import org.briarproject.briar.BriarCoreEagerSingletons
import org.briarproject.briar.BriarCoreModule
import java.security.SecureRandom
import javax.inject.Singleton

@Component(
    modules = [
        BrambleCoreModule::class,
        BrambleJavaModule::class,
        BriarCoreModule::class,
        HeadlessModule::class
    ]
)
@Singleton
internal interface BriarHeadlessApp : BrambleCoreEagerSingletons, BriarCoreEagerSingletons,
    BrambleJavaEagerSingletons, HeadlessEagerSingletons {

    fun getRouter(): Router

    fun getSecureRandom(): SecureRandom
}
